core/src/main/scala/scalaz/StrictTree.scala

```diff
@@ -95,11 +95,23 @@ case class StrictTree[A](
   }
 
   /** Breadth-first traversal. */
-  def levels: Vector[Vector[A]] = {
-    val f = (s: Vector[StrictTree[A]]) => {
-      Foldable[Vector].foldMap(s)((_: StrictTree[A]).subForest)
+  def levels: List[List[A]] = {
+    val f = (s: List[StrictTree[A]]) => {
+      Foldable[List].foldMap(s)((_: StrictTree[A]).subForest)
     }
-    Vector.iterate(Vector(this), size)(f) takeWhile (!_.isEmpty) map (_ map (_.rootLabel))
```

Comment:
The choice of `size` seems to have just been profoundly wrong, it created a `Vector` the size of the entire tree and then shrunk it.